hack/lib/util.sh

```diff
@@ -853,6 +853,20 @@ function kube::util::ensure_dockerized {
   fi
 }
 
+# kube::util::gnu-sed
+# Determines which sed binary is gnu-sed on linux/darwin
+#
+function kube::util::gnu-sed {
+  if LANG=C sed --help 2>&1 | grep -q GNU; then
+    SED="sed"
```

Comment:
hmm, not so nice to set a global var. wdyt?

Reply:
either do this automatically or make kube::util::gnu-sed invoke sed.

Reply:
I actually toyed around with that, but previous implementations all set global vars, and there are some more complex sed commands that are currently set to just consume the var.

You're suggesting just have the function detect, and then invoke the right utility, passing all the args?

Reply:
oh, the other reason I did this is in many scripts, the sed detection intentionally happens earlier.. like in `hack/update-bazel.sh`. If it can't find the right sed and aborts early, there isn't an issue. But if we aborted mid-pipe, the working tree is left in a bad state.

Reply:
The name is odd for the current semantics: `kube::util::gnu-sed`. Don't have a strong opinion which route we go.